apps/financial-aid/web-osk/src/components/AppLayout/AppLayout.tsx

```diff
@@ -11,18 +12,32 @@ import {
 
 import cn from 'classnames'
 import { AppContext } from '@island.is/financial-aid-web/osk/src/components/AppProvider/AppProvider'
+import { Routes } from '@island.is/financial-aid/shared/lib'
 
 interface Props {
   children: ReactNode
 }
 
 const AppLayout = ({ children }: Props) => {
   const { isAuthenticated, user, loadingUser } = useContext(AppContext)
+  const router = useRouter()
 
   useEffect(() => {
     document.title = 'Fjárhagsaðstoð'
   }, [])
 
+  const redirectSpouse = user?.spouse?.hasPartnerApplied && user.spouse.hasFiles
+  const redirectApplicant =
+    !user?.spouse?.hasPartnerApplied && user?.currentApplicationId
+
+  if (
+    router.pathname.startsWith(Routes.application) &&
+    (redirectApplicant || redirectSpouse)
+  ) {
+    router.push(`${Routes.statusPage(user.currentApplicationId as string)}`)
+    return null
+  }
```

Comment:
We already have a similar function returnUrl in pages/index.tsx? Maybe use that and putt it in shared?

Reply:
I decided to move all the redirection in index.tsx into AppLayout.tsx.